modules/schema/validatecreate_test.go

```diff
@@ -0,0 +1,383 @@
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/spaceuptech/space-cloud/config"
+	"github.com/spaceuptech/space-cloud/model"
+	"github.com/spaceuptech/space-cloud/modules/crud"
+)
+
+var queries = `
+ type tweet {
+ 	id: ID @id
+ 	createdAt: DateTime@createdAt
+ 	text: String
+ 	owner: [String]
+ 	location: location@foreign
+	age : Float!
+	isMale:Boolean
+	exp:Integer
+	event:event_logs
+	person : sharad @link(table:sharad, from:Name, to:isMale)
+   }
+
+   type test {
+	id : ID @id
+	person : sharad @link(table:sharad, from:Name, to:isMale)
+   }
+
+   type location {
+ 	id: ID! @id
+ 	latitude: Float
+ 	longitude: Float
+
+   }
+   type sharad {
+ 	  Name : String!
+ 	  Surname : String!
+ 	  age : Integer!
+ 	  isMale : Boolean!
+ 	  dob : DateTime@createdAt
+   }
+   type event_logs {
+ 	id: Integer
+ 	name: String
+   }
+ `
+var Parsedata = config.Crud{
+	"mongo": &config.CrudStub{
+		Collections: map[string]*config.TableRule{
+			"tweet": &config.TableRule{
+				Schema: queries,
+			},
+			"test": &config.TableRule{
+				Schema: queries,
+			},
+			"location": &config.TableRule{
+				Schema: queries,
+			},
+		},
+	},
+}
+
+func TestSchema_ValidateCreateOperation(t *testing.T) {
+
+	tdd := []struct {
+		dbName, coll, description string
+		value                     model.CreateRequest
+		want                      error
+	}{
+		{
+			dbName: "sqlserver",
+			coll:   "tweet",
+			want:   errors.New("No db was found named sqlserver"),
+			value: model.CreateRequest{
+				Document: map[string]interface{}{
+					"male": true,
+				},
+			},
+		},
+		{
+			dbName: "mongo",
+			coll:   "twee",
+			want:   nil,
+			value: model.CreateRequest{
+				Document: map[string]interface{}{
+					"male": true,
+				},
+			},
+		},
+		{
+			dbName: "mongo",
+			coll:   "tweet",
+			want:   errors.New("required field age from collection tweet not present in request"),
+			value: model.CreateRequest{
+				Document: map[string]interface{}{
+					"male": true,
+				},
+			},
+		},
+		{
+			dbName: "mongo",
+			coll:   "tweet",
+			want:   errors.New("invalid document provided for collection (mongo:tweet)"),
+			value: model.CreateRequest{
+				Document: []interface{}{
+					"text", "12PM",
+				},
+			},
+		},
+		{
+			dbName: "mongo",
+			coll:   "tweet",
+			want:   errors.New("required field age from collection tweet not present in request"),
+			value: model.CreateRequest{
+				Document: map[string]interface{}{
+					"isMale": true,
+				},
+			},
+		},
+		{
+			dbName: "mongo",
+			coll:   "location",
+			want:   nil,
+			value: model.CreateRequest{
+				Document: map[string]interface{}{
+					"location": "21",
+				},
+			},
+		},
+	}
+
+	temp := crud.Module{}
+	s := Init(&temp, false)
+	err := s.parseSchema(Parsedata)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, val := range tdd {
+		t.Run(val.description, func(t *testing.T) {
+			err := s.ValidateCreateOperation(val.dbName, val.coll, &val.value)
+			if !reflect.DeepEqual(err, val.want) {
```

Comment:
change this condition to something like this
```suggestion
			if (err != nil) != val.isErrExpected {
```